cmd/diagnostic/PyTorch.go

```diff
@@ -0,0 +1,171 @@
+package diagnostic
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func getPodMachineID(podID, apiKey string) string {
+	url := fmt.Sprintf("https://api.runpod.io/graphql?api_key=%s", apiKey)
+	headers := map[string]string{
+		"Content-Type": "application/json",
+	}
+	query := `
+        query Pod($podId: String!) {
+          pod(input: { podId: $podId }) {
+            machineId
+          }
+        }
+    `
+	data := map[string]interface{}{
+		"query":     query,
+		"variables": map[string]string{"podId": podID},
+	}
+	jsonData, _ := json.Marshal(data)
+
+	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Printf("Failed to fetch machineId: %v\n", err)
+		return ""
+	}
+	defer resp.Body.Close()
+
+	var result map[string]interface{}
+	json.NewDecoder(resp.Body).Decode(&result)
+	if pod, ok := result["data"].(map[string]interface{})["pod"].(map[string]interface{}); ok {
```

Comment:
again, don't use `map[string]any`: just make a struct type that's the shape you want. in this case, it would be 

```go
type Result struct { 
    Data struct  { 
        Pod struct { 
            MachineID string  `json:"machineId"` 
         } `json:"pod"`
    } `json:"data"`
 }
```